staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -450,15 +475,66 @@ type ValidatingAdmissionPolicyBindingSpec struct {
 	ValidationActions []ValidationAction `json:"validationActions,omitempty" protobuf:"bytes,4,rep,name=validationActions"`
 }
 
-// ParamRef references a parameter resource
+// ParamRef describes how to locate the params to be used as input to
+// expressions of rules applied by a policy binding.
+//
+// Parameters may be identified by name and selector.
+//
+// If multiple params are found, they are all evaluated with the policy expressions
+// and the results are ANDed together.
+//
+// If name and `selector` are both specified, then admission will be evaluated
+// on params with the provided `name` OR `selector`. If multiple matches are found
+// all will be evaluated against the policy.
+//
+// A single parameter used for all admission requests can be configurated
+// by setting the `name` field, leaving `selector` blank, and setting namespace
+// if `paramKind` is namespace-scoped.
+//
+// A per-namespace parameter may be used by specifying a namespace-scoped
+// `paramKind` in the policy, setting `name` or `selector` and leaving `namespace`
+// in `paramRef` empty.
+//
+// If all fields are left unset, will match ALL resources of the policy's
+// paramKind. If the paramKind is namespace-scoped, will match ALL
+// params in the request object's namespace (or `default` is the request object
+// is cluster-scoped)
+//
 // +structType=atomic
 type ParamRef struct {
 	// Name of the resource being referenced.
+	//
+	// +optional
 	Name string `json:"name,omitempty" protobuf:"bytes,1,rep,name=name"`
-	// Namespace of the referenced resource.
-	// Should be empty for the cluster-scoped resources
+
+	// Namespace of the referenced resource. Allows limiting the search for params
+	// to a specific namespace. Applies to both `name` and `selector` fields.
+	//
+	// - If `paramKind` is cluster-scoped, this field MUST be unset. Setting this
+	// field results in a configuration error.
+	// - If `paramKind` is namespace-scoped, the namespace of the object being
+	// evaluated for admission will be used when this field is left unset; or
+	// `default` if the request object is cluster-scoped.
+	//
 	// +optional
 	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,rep,name=namespace"`
+
+	// Selector can be used to match multiple param objects based on their labels.
+	//
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,3,rep,name=selector"`
+
+	// ParameterNotFoundAction controls the behavior of the binding when the resource
+	// exists, and name or selector is valid, but there are no parameters
+	// matched by the binding. If the value is set to `Allow`, then no
+	// matched parameters will be treated as successful validation by the binding.
+	// If set to `Deny`, then no matched parameters will be subject to the
+	// `failurePolicy` of the policy.
+	//
+	// Allowed values are `Allow` or `Deny`
+	// Default to `Deny`
```

Comment:
For v1alpha1 it's backwards compatible to default to `Deny`. For internal & v1beta1 types it will have no default.

context: https://github.com/kubernetes/kubernetes/pull/119215#discussion_r1265838722